pkg/kratos/service.go

```diff
@@ -123,14 +125,21 @@ func (s *Service) CreateBrowserLoginFlow(
 	ctx, span := s.tracer.Start(ctx, "kratos.Service.CreateBrowserLoginFlow")
 	defer span.End()
 
-	flow, resp, err := s.kratos.FrontendApi().
+	request := s.kratos.FrontendApi().
 		CreateBrowserLoginFlow(ctx).
 		Aal(aal).
 		ReturnTo(returnTo).
-		LoginChallenge(loginChallenge).
 		Refresh(refresh).
-		Cookie(httpHelpers.CookiesToString(cookies)).
-		Execute()
+		Cookie(httpHelpers.
+			CookiesToString(cookies))
+
+	if loginChallenge != "" {
+		request = request.LoginChallenge(loginChallenge)
+	} else if returnTo == "" {
```

Comment:
is this `loginChallenge is "abc" and returnTo is not set`?

Reply:
No, it's when `loginChallenge == "" && returnTo == ""`

Reply:
as discussed probably worth being more explicit with another separate `if`